python/ray/autoscaler/gcp/config.py

```diff
@@ -104,16 +102,37 @@ def generate_rsa_key_pair():
     return public_key, pem
 
 
+def create_crm_client(config, gcp_credentials):
+    return discovery.build(
+        "cloudresourcemanager", "v1", credentials=gcp_credentials)
+
+
+def create_iam_client(config, gcp_credentials):
+    return discovery.build("iam", "v1", credentials=gcp_credentials)
+
+
+def create_compute_client(config, gcp_credentials):
+    return discovery.build("compute", "v1", credentials=gcp_credentials)
+
+
 def bootstrap_gcp(config):
-    config = _configure_project(config)
-    config = _configure_iam_role(config)
-    config = _configure_key_pair(config)
-    config = _configure_subnet(config)
+    service_account_info = config["provider"].get("gcp_credentials")
```

Comment:
Does the ray-schema.json schema file need to be updated?

Reply:
Wasn't too sure about this... there is nothing in the schema file for aws_credentials, which I used as a template for this change. @thomasdesr thoughts?

Reply:
Lets go ahead and fix this particular case.

`ray-schema.json` is pretty out of sync atm but that is a separate problem we should probably fix.

Reply:
NOte: aws provider has the same issue which is discussed over here https://github.com/ray-project/ray/issues/8420